src/model/task/util.ts

```diff
@@ -5,7 +5,11 @@ import { DeepPartial } from "utility-types";
 import { DEFAULT_PRIORITY_VALUE, DEFAULT_TYPE_VALUE, Task, TASK_WITH_DEFAULT_VALUES } from "@/model/task/schema";
 
 /**
- * @param parts - the task parts to merge.
+ * Merges multiple partial Task objects into a complete Task using a custom merge strategy.
+ *
+ * The function begins with default Task values and sequentially merges each provided partial Task. For each property, the front-most non-default value is retained. Special handling is applied for properties such as task type, priority, strings, DateTime objects, and Set collections.
+ *
+ * @param parts - The partial Task objects to merge.
```

Comment:
```suggestion
 * @param parts - The partial Task objects to merge.
```
The project's JSDoc ESLint rules flags extra lines between tags as errors.